pilot/pkg/config/kube/gateway/controller.go

```diff
@@ -185,6 +189,10 @@ func (c *Controller) Reconcile(ps *model.PushContext) error {
 	referenceGrant := c.cache.List(gvk.ReferenceGrant, metav1.NamespaceAll)
 	serviceEntry := c.cache.List(gvk.ServiceEntry, metav1.NamespaceAll) // TODO lazy load only referenced SEs?
 
+	gateway = slices.FilterInPlace(gateway, func(gw config.Config) bool {
```

Comment:
hmmm, what is the longest interval we will go between pushes?  Could this result in a scenario where the tags change after the latest pushcontext to include more gateways, but these gateways are only given the latest pushcontext when they connect to xds, and are effectively stale until a new pushcontext is kicked off?

Reply:
In theory forever. We could do `tagwatcher.AddHandler({push xds})`?